python/ray/data/_internal/stats.py

```diff
@@ -165,6 +167,9 @@ def __init__(self, max_stats=1000):
         # Cache of calls to ray.nodes() to prevent unnecessary network calls
         self._ray_nodes_cache: Dict[str, str] = {}
 
+        # Initialize the metadata exporter
+        self._metadata_exporter = LoggerDataMetadataExporter.create_if_enabled()
```

Comment:
this file should avoid directly referencing LoggerDataMetadataExporter
We can have a `get_data_metadata_exporter` function that returns a `DataMetadataExporter`